Refactor the branding documentation (#73)
owncloud/docs-client-branding: modules/ROOT/pages/ios_app/preparation/provisioning_profiles.adoc

= Create Provisioning Profiles
:toc: right
:description: A provisioning profile is a collection of information that links an App ID with signing certificates and authorized devices.

== Introduction

{description} It is used to control and authorize the devices on which the app can run and the Apple services it can access.

== Overview

Eight provisioning profiles need to be created:

> * 4 Ad Hoc and

Reviewer: ```suggestion
* 4 ad hoc and
```

> * 4 App Store.

Reviewer: ```suggestion
* 4 app store.
```

> You will email the four Ad Hoc profiles, and your P12 certificate (which you will create after your provisioning profiles), to support@owncloud.com after building your branded app with the ownBrander app on {customer-owncloud-url}[customer.owncloud.com]. *Do not send us the App Store profiles.* All eight of these profiles must be stored on your Mac PC.

Reviewer: ```suggestion
You will email the four ad hoc profiles and your P12 certificate (which you will create after your provisioning profiles), to support@owncloud.com after building your branded app with the ownBrander app on {customer-owncloud-url}[customer.owncloud.com]. *Do not send us the app store profiles.* All eight of these profiles must be stored on your Mac PC.
```

== Ad Hoc Provisioning Profiles

=== First Profile

Go to menu:Provisioning Profiles[All], then click the btn:[plus button] at the top right to open the _Add iOS Provisioning Profile_ screen. Select btn:[Ad Hoc] and click btn:[Continue].

image::ios_app/preparation/provisioning_profiles/cert-35.png[image, width=450]

On the *Select App ID* screen select the first of the three App IDs that you created and click btn:[Continue]. (The first one has the shortest name, if you followed the naming conventions in this manual.)

image::ios_app/preparation/provisioning_profiles/cert-36.png[image, width=450]

Select the certificate that you created at the beginning of this process and click btn:[Continue].

image::ios_app/preparation/provisioning_profiles/cert-38.png[image, width=450]

Select the devices that you want to install and test your app on, then click btn:[Continue].

image::ios_app/preparation/provisioning_profiles/cert-39.png[image, width=450]

Name your provisioning profile with a descriptive *Profile Name* and click btn:[Generate].

image::ios_app/preparation/provisioning_profiles/cert-40.png[image, width=450]

> When it has generated, download your new profile to your Mac computer.

Reviewer: ```suggestion
When it has been generated, download your new profile to your Mac computer.
```

=== Fourth Profile

Return to the "Your provision profile is ready" screen, scroll to the bottom and click btn:[Add Another]. On the following screen select btn:[Ad Hoc] and click btn:[Continue].

image::ios_app/preparation/provisioning_profiles/cert-35.png[image, width=450]

This time select the *.ShareExtApp* app ID and click btn:[Continue].

image::ios_app/preparation/provisioning_profiles/cert-46.png[image, width=450]

Select the certificate that you created at the beginning of this process and click btn:[Continue].

image::ios_app/preparation/provisioning_profiles/cert-43.png[image, width=450]

Select the devices that you want to install and test your app on, then click btn:[Continue]. These must be the same devices you selected for the first provisioning profile.

image::ios_app/preparation/provisioning_profiles/cert-39.png[image, width=450]

Give this provisioning profile the same name as your first profile plus *.ShareExtApp* and click btn:[Generate].
There is a 50-character limit, but don’t worry about counting characters because it will be automatically truncated if you go over.

image::ios_app/preparation/provisioning_profiles/cert-58.png[image, width=450]

> Download it to your Mac computer, and then double-click to install it in Xcode. You should now see all of your Ad Hoc provisioning profiles listed in your "iOS Provisioning Profiles".

Reviewer: ```suggestion
Download it to your Mac computer, and then double-click to install it in Xcode. You should now see all of your ad hoc provisioning profiles listed in your "iOS Provisioning Profiles".
```

image::ios_app/preparation/provisioning_profiles/cert-59.png[image, width=450]

== App Store Provisioning Profiles

> Creating your four App Store profiles is the same as creating your Ad Hoc profiles, except that when you start you check the App Store checkbox, and you won’t select testing devices.

Reviewer: ```suggestion
Creating your four app store profiles works the same way as creating your ad hoc profiles, except that when you start you check the App Store checkbox, and you won’t select testing devices.
```

image::ios_app/preparation/provisioning_profiles/cert-62.png[image, width=450]

> When you’re finished, you’ll have eight new provisioning profiles. Remember, when you build your app on ownBuilder you only send in the four Ad Hoc profiles, plus your P12 certificate.

Reviewer: ```suggestion
When you’re finished, you’ll have eight new provisioning profiles. Remember, when you build your app on ownBuilder, you should only send the four ad hoc profiles, plus your P12 certificate.
```
